<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>783</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="title">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>30</y>
      <width>381</width>
      <height>61</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 28pt &quot;Adobe Devanagari&quot;;</string>
    </property>
    <property name="text">
     <string>EIGHT PUZZLE</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_start">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>130</y>
      <width>191</width>
      <height>211</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Please input start state 请输入起始状态&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="title">
     <string>   start 开始状态</string>
    </property>
    <widget class="QLineEdit" name="start_0_0">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="start_0_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="start_0_2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="start_1_0">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="start_1_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>90</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="start_1_2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>90</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="start_2_0">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="start_2_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>150</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="start_2_2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>150</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_end">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>130</y>
      <width>191</width>
      <height>211</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Please input end state 请输入终止状态&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="title">
     <string>   end 目标状态</string>
    </property>
    <widget class="QLineEdit" name="end_0_0">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="end_0_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="end_0_2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>30</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="end_1_0">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="end_1_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>90</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="end_1_2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>90</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="end_2_0">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="end_2_1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>150</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="end_2_2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>150</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QComboBox" name="hnSelect">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>370</y>
      <width>381</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Please choose a h(n) function 请选择h(n)函数&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <item>
     <property name="text">
      <string>not in the target location chess number</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Manhattan Distance</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Euclidean distance</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>370</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>h(n)：</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>660</y>
      <width>82</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="MoreButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;click here to show search tree&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>more</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="GoButton">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>430</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Go</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_process">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>480</y>
      <width>201</width>
      <height>211</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;show moving process&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="title">
     <string>process 过程演示</string>
    </property>
    <widget class="QWidget" name="layoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>181</width>
       <height>171</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_0_1">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_0_0">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_1_0">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_2_0">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_1_1">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_1_2">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_0_2">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_2_1">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2" alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="process_2_2">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QLabel" name="StepLabel">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>530</y>
      <width>341</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Total steps</string>
    </property>
   </widget>
   <widget class="QLabel" name="TimeLabel">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>560</y>
      <width>341</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Total time</string>
    </property>
   </widget>
   <widget class="QLabel" name="GnumLabel">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>590</y>
      <width>331</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate node number</string>
    </property>
   </widget>
   <widget class="QLabel" name="ExnumLabel">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>620</y>
      <width>341</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Extend node number</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHELP">
    <property name="title">
     <string>HELP</string>
    </property>
    <addaction name="Introduction"/>
    <addaction name="Usage"/>
   </widget>
   <addaction name="menuHELP"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="Introduction">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>What is an EightDigital game?</string>
   </property>
  </action>
  <action name="Usage">
   <property name="text">
    <string>How to use this program?</string>
   </property>
  </action>
  <action name="A_algorithm">
   <property name="text">
    <string>A* algorithm</string>
   </property>
  </action>
  <action name="More">
   <property name="text">
    <string>more</string>
   </property>
  </action>
  <action name="A_algorithm_2">
   <property name="text">
    <string>A* algorithm</string>
   </property>
  </action>
  <action name="More_2">
   <property name="text">
    <string>More</string>
   </property>
  </action>
  <action name="More_4">
   <property name="text">
    <string>More</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
